### ISSUE TITLE
Transfer of resource shows wrong owner in resource overview?

### ISSUE THREAD

**Issue by sbocconi** [reflow-project/weloop#101]
I have created a resource that was transferred from some user to me, as I wanted to describe the fact that I was not the producer or the original owner, but I bought it.

In the overview page, the owner is the provider of the resource, not the receiver, but being a transfer, I should be the owner, or not?

https://weloop.reflowproject.eu/resource/01FCT51592SSVQG8WC64GM5ZYC

The location is also Amsterdam, but the map shows Barcelona.

**Comment by sbocconi** [comment]
Additionally, the units available for my resource were only liters and kilos. What about unit:Number used for objects that are treated as single pieces, like guitars, TVs, computers, etc?

**Comment by viktorsmari** [comment]
Good question @sbocconi 

Right now (on the staging / test server) we have a multiple test units, many are duplicates.

Ideally we need to clean the data on the API server, remove the duplicates, and add missing ones.
I did find this unit which would fit your guitar example - nice guitar by the way! :guitar: :man_singer:

```javascript
       {
          "id": "01F7XYVV2YXYZDV4FFC9812TNB",
          "label": "pieces",
          "symbol": "pcs."
        },
```

Using the [explore](http://api.reflowproject.eu/api/explore), I can find multiple units, and most of them are `liter` or `kilo`

![2021-08-11_11-18-39](https://user-images.githubusercontent.com/1689020/129003715-1d6778d9-d1d6-4f2e-b2fd-c8cd67f6af56.png)


```javascript
{
  unitsPages(limit:30, before: "01EVTV5Q0BAJHJ7AZG5TRFRKV3") {
  	edges {
    	id
    	label
        symbol
    }
  }
}
```

I think one of the issue is that by default it returns only 5 units, we have to specify `limit:30` to get more. 30 is the maximum per page.
So we need to be able to **paginate** if we have more than 30 (which we do in this case including duplicates).
We have run into the pagination issues in [more issues](is:issue paginate ) in the past. In the future we might need to be able to **search** all `units` if we have many. I don't think the API supports searching `units` at this point.


Here is an example on how to create a unit from the tour docs:
https://reflowos.dyne.org/docs/api_tour#create-a-unit

**Comment by sbocconi** [comment]
thanks for your reply @viktorsmari !

> Ideally we need to clean the data on the API server, remove the duplicates, and add missing ones.

I agree, creating units seems to be an administrative role and not left to users, right?

>We have run into the pagination issues in more issues in the past. In the future we might need to be able to search all units if we have many. I don't think the API supports searching units at this point.

This seems like a good idea, should an issue be open against RelowOS?

> nice guitar by the way! 🎸 👨‍🎤

Thanks, I am going to add a picture as well!

@viktorsmari, what about the fact that I am not the owner?

And the map showing Barcelona iso Amsterdam?

**Comment by viktorsmari** [comment]
> I agree, creating units seems to be an administrative role and not left to users, right?

I was able to create a new `unit` today with my user, and I think I am a **normal** user - It looks like everyone can create units?  :thinking: 
I also think this should be an admin only feature like you said. Something for the API team to look at?

I created a new issue for the `units` in #102 - asking for more than 5 to be shown, and also if it's possible to search / filter. That might be an API issue and not a frontend issue.

Here is one issue we opened for the API repository about searching and paging:
- https://github.com/dyne/zenpub/issues/52#issuecomment-843217417

> And the map showing Barcelona iso Amsterdam?

I have the same issue when I create a new resource, the location is correct for the resource ("Vestmannaeyjar"), but the location for the map shown is also Barcelona
https://weloop.reflowproject.eu/resource/01FCTJP9RBRAQAGSBZ3TYPYXEG

You can also see I am both the provider and receiver of that event

**Comment by pral2a** [comment]
Related to https://github.com/reflow-project/weloop/issues/107#issuecomment-955007353 potentially fixed via #149
